src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -133,12 +184,32 @@ public SubmissionDto submitSolution(String roomId, SubmissionRequest request) {
     }
 
     // Send a notification through a socket update.
-    public NotificationDto sendNotification(List<String> userIdList) {
+    public GameNotificationDto sendNotification(String roomId, GameNotificationDto notificationDto) {
+        Game game = getGameFromRoomId(roomId);
+
         /**
-         * TODO: Get the players from the userIdList,
-         * receive a game notification with details.
+         * The notification type must be present.
+         * 
+         * If the initiator exists on the notificationDto (not required),
+         * then ensure that they exist in the room.
+         * 
+         * If initiator does not exist, ensure that the notification type 
+         * does not require an initiator.
          */
-        return notificationService.sendNotification(GameNotification.SUBMIT_CORRECT, new ArrayList<>());
+        if (notificationDto.getNotificationType() == null) {
+            throw new ApiException(GameError.EMPTY_FIELD);
+        } else if (notificationDto.getInitiator() != null
+            && !game.getPlayers().containsKey(notificationDto.getInitiator().getUserId())) {
+            throw new ApiException(GameError.USER_NOT_IN_GAME);
+        } else if (notificationDto.getInitiator() == null
+            && Utility.initiatorNotifications.contains(notificationDto.getNotificationType())) {
+            throw new ApiException(GameError.NOTIFICATION_REQUIRES_INITIATOR);
+        } else if (notificationDto.getContent() == null
+            && Utility.contentNotifications.contains(notificationDto.getNotificationType())) {
+            throw new ApiException(GameError.NOTIFICATION_REQUIRES_CONTENT);       
+        }
```

Comment:
This is a personal preference so feel free to just resolve and ignore this:

nit: Change `else if` to `if` on new lines, since the else if is redundant (I feel like this helps with readability but again it's by personal preference so it's cool if you ignore)

Reply:
Sure I'm fine with that -- updated!

(I think I told you the exact opposite on another PR lol, to each their own -- it's not a huge preference by me though so I'll go by this from now on.)